packages/zoe/src/contracts/sellItems.js

```diff
@@ -21,107 +27,100 @@ import '../../exported';
  * available to sell, and the money should be pricePerItem times the number of
  * items requested.
  *
- * @param {ContractFacet} zcf
+ * @type {ContractStartFn}
  */
-const makeContract = zcf => {
-  const allKeywords = ['Items', 'Money'];
+const start = zcf => {
   const {
-    assertKeywords,
-    rejectOffer,
-    checkHook,
-    assertNatMathHelpers,
-    trade,
-  } = makeZoeHelpers(zcf);
-  assertKeywords(harden(allKeywords));
-
-  const { pricePerItem } = zcf.getInstanceRecord().terms;
-  assertNatMathHelpers(pricePerItem.brand);
-  let sellerOfferHandle;
-
-  const sellerOfferHook = offerHandle => {
-    sellerOfferHandle = offerHandle;
+    pricePerItem,
+    issuers,
+    maths: { Money: moneyMath, Items: itemsMath },
+  } = zcf.getTerms();
+  const allKeywords = ['Items', 'Money'];
+  assertIssuerKeywords(zcf, harden(allKeywords));
+  assertUsesNatMath(zcf, pricePerItem.brand);
+
+  let sellerSeat;
+
+  const sell = seat => {
+    sellerSeat = seat;
     return defaultAcceptanceMsg;
   };
 
-  const buyerOfferHook = buyerOfferHandle => {
-    const { brandKeywordRecord } = zcf.getInstanceRecord();
-    const [sellerAllocation, buyerAllocation] = zcf.getCurrentAllocations(
-      [sellerOfferHandle, buyerOfferHandle],
-      [brandKeywordRecord, brandKeywordRecord],
-    );
-    const currentItemsForSale = sellerAllocation.Items;
-    const providedMoney = buyerAllocation.Money;
-
-    const { proposal } = zcf.getOffer(buyerOfferHandle);
-    const wantedItems = proposal.want.Items;
-    const numItemsWanted = wantedItems.value.length;
-    const totalCostValue = pricePerItem.value * numItemsWanted;
-    const moneyAmountMaths = zcf.getAmountMath(pricePerItem.brand);
-    const itemsAmountMath = zcf.getAmountMath(wantedItems.brand);
+  const buy = buyerSeat => {
+    const currentItemsForSale = sellerSeat.getAmountAllocated('Items');
+    const providedMoney = buyerSeat.getAmountAllocated('Money');
```

Comment:
You could, but in the past, people haven't understood the difference between the proposal and the current allocation, so I think we should use `getAmountAllocated` if we are trying to get the current allocation to not teach people to use the proposal, since they will probably use it when we don't have those guarantees